WordPress/WordPressTest/PostRepositoryTests.swift

```diff
@@ -0,0 +1,140 @@
+        let accountID = accountService.createOrUpdateAccount(withUsername: "username", authToken: "token")
+        try accountService.setDefaultWordPressComAccount(XCTUnwrap(mainContext.existingObject(with: accountID) as? WPAccount))
+
+        let blog = try BlogBuilder(mainContext).withAccount(id: accountID).build()
+
+        contextManager.saveContextAndWait(mainContext)
+
+        blogID = .init(saved: blog)
+        remoteMock = PostServiceRESTMock()
+        let remoteFactory = PostServiceRemoteFactoryMock()
+        remoteFactory.remoteToReturn = remoteMock
+        repository = PostRepository(coreDataStack: contextManager, remoteFactory: remoteFactory)
+    }
+
+    func testGetPost() async throws {
+        let post = RemotePost(siteID: 1, status: "publish", title: "Post: Test", content: "This is a test post")
+        post?.type = "post"
+        remoteMock.remotePostToReturnOnGetPostWithID = post
+        let postID = try await repository.getPost(withID: 1, from: blogID)
+        let isPage = try await contextManager.performQuery { try $0.existingObject(with: postID) is Page }
+        let title = try await contextManager.performQuery { try $0.existingObject(with: postID).postTitle }
+        let content = try await contextManager.performQuery { try $0.existingObject(with: postID).content }
+        XCTAssertFalse(isPage)
+        XCTAssertEqual(title, "Post: Test")
+        XCTAssertEqual(content, "This is a test post")
+    }
+
+    func testGetPage() async throws {
+        let post = RemotePost(siteID: 1, status: "publish", title: "Post: Test", content: "This is a test post")
+        post?.type = "page"
+        remoteMock.remotePostToReturnOnGetPostWithID = post
+        let postID = try await repository.getPost(withID: 1, from: blogID)
+        let isPage = try await contextManager.performQuery { try $0.existingObject(with: postID) is Page }
+        let title = try await contextManager.performQuery { try $0.existingObject(with: postID).postTitle }
+        let content = try await contextManager.performQuery { try $0.existingObject(with: postID).content }
+        XCTAssertTrue(isPage)
+        XCTAssertEqual(title, "Post: Test")
+        XCTAssertEqual(content, "This is a test post")
+    }
+
+}
+
+// These mock classes are copied from PostServiceWPComTests. We can't simply remove the `private` in the original class
+// definition, because Xcode would complian about 'WordPress' module not found.
+
+private class PostServiceRemoteFactoryMock: PostServiceRemoteFactory {
+    var remoteToReturn: PostServiceRemote?
+
+    override func forBlog(_ blog: Blog) -> PostServiceRemote? {
+        return remoteToReturn
+    }
+
+    override func restRemoteFor(siteID: NSNumber, context: NSManagedObjectContext) -> PostServiceRemoteREST? {
+        return remoteToReturn as? PostServiceRemoteREST
+    }
+}
+
+private class PostServiceRESTMock: PostServiceRemoteREST {
```

Comment:
Have you considered making this a type conforming to `PostServiceRemote` instead of a `PostServiceRemoteREST` subclass?

Reply:
No, I didn't look too much into this class's implementation. It is copied from [PostServiceWPComTests](https://github.com/wordpress-mobile/WordPress-iOS/blob/af331d74b5f3209cc8a95cca9ac656ef39e1545f/WordPress/WordPressTest/Services/PostServiceWPComTests.swift#L373) 😅